src/libraries/Common/tests/System/Security/Cryptography/AlgorithmImplementations/SlhDsa/SlhDsaTests.cs

```diff
@@ -0,0 +1,83 @@
+            Assert.Throws<ArgumentNullException>(() => SlhDsa.ImportSlhDsaPrivateSeed(null, ReadOnlySpan<byte>.Empty));
+        }
+
+        [Theory]
+        [MemberData(nameof(AlgorithmsData))]
+        public static void ArgumentValidation(SlhDsaAlgorithm algorithm)
+        {
+            using SlhDsa slhDsa = SlhDsa.GenerateKey(algorithm);
+
+            Assert.Throws<ArgumentNullException>(() => slhDsa.ExportEncryptedPkcs8PrivateKey(ReadOnlySpan<byte>.Empty, null));
+            Assert.Throws<ArgumentNullException>(() => slhDsa.ExportEncryptedPkcs8PrivateKeyPem(ReadOnlySpan<byte>.Empty, null));
+            Assert.Throws<ArgumentNullException>(() => slhDsa.ExportEncryptedPkcs8PrivateKeyPem(ReadOnlySpan<char>.Empty, null));
+            Assert.Throws<ArgumentNullException>(() => slhDsa.TryExportEncryptedPkcs8PrivateKey(ReadOnlySpan<byte>.Empty, null, Span<byte>.Empty, out _));
+            Assert.Throws<ArgumentNullException>(() => slhDsa.TryExportEncryptedPkcs8PrivateKey(ReadOnlySpan<char>.Empty, null, Span<byte>.Empty, out _));
+
+            int publicKeySize = algorithm.PublicKeySizeInBytes;
+            int secretKeySize = algorithm.SecretKeySizeInBytes;
+            int privateSeedSize = algorithm.SecretKeySizeInBytes / 4 * 3;
+
+            Assert.Throws<CryptographicException>(() => SlhDsa.ImportSlhDsaPublicKey(algorithm, new byte[publicKeySize - 1]));
+            Assert.Throws<CryptographicException>(() => SlhDsa.ImportSlhDsaPublicKey(algorithm, new byte[publicKeySize + 1]));
+            Assert.Throws<CryptographicException>(() => SlhDsa.ImportSlhDsaSecretKey(algorithm, new byte[secretKeySize - 1]));
+            Assert.Throws<CryptographicException>(() => SlhDsa.ImportSlhDsaSecretKey(algorithm, new byte[secretKeySize + 1]));
+            Assert.Throws<CryptographicException>(() => SlhDsa.ImportSlhDsaPrivateSeed(algorithm, new byte[privateSeedSize - 1]));
+            Assert.Throws<CryptographicException>(() => SlhDsa.ImportSlhDsaPrivateSeed(algorithm, new byte[privateSeedSize + 1]));
+
+            Assert.Throws<ArgumentException>(() => slhDsa.ExportSlhDsaPublicKey(new byte[publicKeySize - 1]));
+            Assert.Throws<ArgumentException>(() => slhDsa.ExportSlhDsaSecretKey(new byte[secretKeySize - 1]));
+            Assert.Throws<ArgumentException>(() => slhDsa.ExportSlhDsaPrivateSeed(new byte[privateSeedSize - 1]));
+
+            // Context length must be less than 256
+            Assert.Throws<ArgumentOutOfRangeException>(() => slhDsa.SignData(ReadOnlySpan<byte>.Empty, Span<byte>.Empty, new byte[256]));
+            Assert.Throws<ArgumentOutOfRangeException>(() => slhDsa.VerifyData(ReadOnlySpan<byte>.Empty, Span<byte>.Empty, new byte[256]));
+        }
+
+        [Theory]
+        [MemberData(nameof(AlgorithmsData))]
+        public static void AlgorithmMatches(SlhDsaAlgorithm algorithm)
+        {
+            using SlhDsa slhDsa = SlhDsa.GenerateKey(algorithm);
+            Assert.Equal(algorithm, slhDsa.Algorithm);
```

Comment:
OK.  I lied, I'm mentioning it again, but only to disagree with myself.

This test checks something on the instance, but really it's testing the static method GenerateKey.  So it's fine where it is :smile:

Another reasonable test for GenerateKey is that it returns instances that are not castable to any other public type.  E.g.

```C#
Type t = instance.GetType();

while (t != null && t != typeof(SlhDsa))
{
    Assert.False(t.IsPublic);
    t = t.BaseClass;
}
```

(written off the top of my head, that probably doesn't compile)